core/src/test/java/nl/nn/adapterframework/doc/model/FrankDocModelTest.java

```diff
@@ -0,0 +1,201 @@
+package nl.nn.adapterframework.doc.model;
+
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import nl.nn.adapterframework.doc.ModelBuilder;
+
+public class FrankDocModelTest {
+	private static final String IBISDOCREF = "nl.nn.adapterframework.doc.testtarget.ibisdocref";
+	private static final String REFERRED_CHILD = IBISDOCREF + ".ChildTarget";
+	private static final String REFERRED_PARENT = IBISDOCREF + ".ParentTarget";
+
+	private FrankDocModel instance;
+
+	@Before
+	public void setUp() {
+		instance = new FrankDocModel();
+	}
+
+	@Test
+	public void whenSetterAndGetterThenAttribute() {
+		checkReflectAttributeCreated("attributeSetterGetter");
+	}
+
+	private FrankAttribute checkReflectAttributeCreated(String attributeName) {
+		Map<String, FrankAttribute> actual = getReflectInvestigatedFrankAttributes();
+		Assert.assertTrue(actual.containsKey(attributeName));
+		Assert.assertEquals(attributeName, actual.get(attributeName).getName());
+		return actual.get(attributeName);
+	}
+
+	private Map<String, FrankAttribute> getReflectInvestigatedFrankAttributes() {
+		return getAttributesOfClass("nl.nn.adapterframework.doc.testtarget.reflect.FrankAttributeTarget");
+	}
+
+	/**
+	 * @param className The name of the Class<?> to get FrankAttribute objects for.
+	 * @return The FrankAttributes that ModelBuilder can produce from a Class<?>. Note
+	 * that the describingElement is not set, because we are not testing here in relation
+	 * to creating the enclosing FrankElement of the attributes.
```

Comment:
Ik vind het prettiger om een javadoc te lezen die beschrijft wat de method doet, in plaats van dat te moeten reconstrueren uit de \@param en \@return tag